src/cli/msig/index.ts

```diff
@@ -163,6 +165,66 @@ export default async function msig(args: string[]) {
                 newGovernanceAccounts
             );
 
+            if (test) await executeProposal(proposer, proposalName, proposalHash);
+        } else if (proposalType === 'deploy-contract') {
+            const contractName = args[3];
+
+            if (!contractName) {
+                throw new Error('Contract name must be provided for deploy-contract proposal');
+            }
+
+            const contractInfo = {
+                account: contractName,
+                contractDir: path.join(__dirname, `../../Tonomy-Contracts/contracts/${contractName}`),
+            };
+            const action = await deployContract(contractInfo, signer, { throughTonomyProxy: true, actionData: true });
+
+            if (action) {
+                const proposalHash = await createProposal(
+                    proposer,
+                    proposalName,
+                    [action],
+                    privateKey,
+                    newGovernanceAccounts
+                );
+
+                if (test) await executeProposal(proposer, proposalName, proposalHash);
+            }
+        } else if (proposalType === 'eosio.code-permission') {
+            const ownerAuthority = Authority.fromAccount({ actor: 'vesting.tmy', permission: 'owner' });
```

Comment:
First it should look up the account,

then, for both the `owner` and `active` permission it should add the additional codePermission to it to whatever permission already exists.

In this way, It will preserve the existing permissions, however they are while supplementing them with the necessary authorizations with `.addCodePermission('vesting.tmy');`